### ISSUE TITLE
ODE example on tensorflow 2.2.0

### ISSUE THREAD

**Issue by Jhsmit** [henrysky/astroNN#9]
When I run the ``odeint`` example on tensorflow 2.2.0 i get the error:

```
  File "C:\Users\jhsmi\pp\astroNN\astroNN\neuralode\dop853.py", line 177, in dopri853core
    if tf.equal(hmax, 0.0):
  File "C:\Users\jhsmi\Miniconda3\envs\py37_tf_dev\lib\site-packages\tensorflow\python\framework\ops.py", line 778, in __bool__
    self._disallow_bool_casting()
  File "C:\Users\jhsmi\Miniconda3\envs\py37_tf_dev\lib\site-packages\tensorflow\python\framework\ops.py", line 545, in _disallow_bool_casting
    "using a `tf.Tensor` as a Python `bool`")
  File "C:\Users\jhsmi\Miniconda3\envs\py37_tf_dev\lib\site-packages\tensorflow\python\framework\ops.py", line 532, in _disallow_when_autograph_enabled
    " decorating it directly with @tf.function.".format(task))
tensorflow.python.framework.errors_impl.OperatorNotAllowedInGraphError: using a `tf.Tensor` as a Python `bool` is not allowed: AutoGraph did not convert this function. Try decorating it directly with @tf.function.

```

It works fine for me on TF 2.1.0

**Comment by henrysky** [comment]
Are you using the latest master branch? And which example are you talking about? I am currently using TF2.2 and examples on the documentation seems to be working fine for me.

**Comment by Jhsmit** [comment]
Yes, latest master branch (0b0e34d)

I'm referring to the sin integration example:
https://astronn.readthedocs.io/en/latest/neuralnets/neuralODE.html

I've rechecked with clean conda env, python 3.7 on windows, installed tensorflow with ``pip install tensorflow==2.2``, and the error is now gone. Probably I broke something in my old env.

TF (+gpu) 2.3.0 again seems to break things but thats another issue.
